app/Libraries/Search/BeatmapsetSearch.php

```diff
@@ -398,6 +405,17 @@ private function addTextFilter(BoolQuery $query, string $paramField, array $fiel
         $query->must($subQuery);
     }
 
+    private function addTagsFilter(BoolQuery $query): void
+    {
+        if (!present($this->params->tags)) {
+            return;
+        }
+
+        foreach ($this->params->tags as $tag) {
+            $query->filter(QueryHelper::queryString($tag, ['beatmaps.top_tags'], 'and'));
```

Comment:
apparently searching for `tag="a b"` will also include beatmaps which has tag `a` and `b` in addition to `a b`...? not sure if intentional.

also `tag="a"` will also include `a b` and I can't find way to filter just `a` (although there isn't such tag at the moment)

Reply:
Well that's partly a parsing issue unless we want to start going `tag=""a""` for exact matches...
`creator="a"` already includes `a b` and `a c` so it's not really doing anything different at the moment.

Reply:
`creator=a` does only include user `a` though 🤔 (it uses user id lookup if the user exists)